controlsfx/src/test/java/org/controlsfx/control/tableview2/TableView2Test.java

```diff
@@ -0,0 +1,303 @@
+                                });
+                        return rowTableRow2;
+                    });
+                });
+        List<TableColumn2<RowItem, String>> fixed = new ArrayList<>();
+        tableView.getColumns()
+                .setAll(IntStream.range(0, NUMBER_OF_COLUMNS)
+                        .mapToObj(index -> {
+                            TableColumn2<RowItem, String> column = new TableColumn2<>();
+                            column.setId("column_" + index);
+                            column.setText("#" + index);
+                            column.setCellValueFactory(param -> param.getValue().values[index]);
+                            if (index < 3) {
+                                fixed.add(column);
+                            }
+                            final Callback<TableColumn<RowItem, String>, TableCell<RowItem, String>> cellFactory = column.getCellFactory();
+                            column.setCellFactory(cellFactory);
+                            return column;
+                        })
+                        .collect(Collectors.toList()));
+        tableView.getFixedColumns()
+                .setAll(fixed);
+        data = FXCollections.observableArrayList();
+        tableView.setItems(data);
+        numberRowChildrenModifications = new AtomicLong();
+    }
+
+    private void fillTableData() {
+        data.setAll(IntStream.range(0, NUMBER_OF_ROWS)
+                .mapToObj((int rowIndex) -> new RowItem(rowIndex, NUMBER_OF_COLUMNS))
+                .collect(Collectors.toList()));
+    }
+
+    private void setupManipulators() {
+        dataManipulationCountdown = new CountDownLatch(1);
+        allowedToRun = new AtomicBoolean(true);
+        for (int i = 0; i < THREADS; i++) {
+            this.dataManipulators.submit(() -> {
+                Random random = new SecureRandom();
+
+                try {
+                    dataManipulationCountdown.await();
+                } catch (InterruptedException e) {
+                    // return silently
+                    return;
+                }
+                while (allowedToRun.get()) {
+                    RowItem rowItem = tableView.getItems()
+                            .get(random.nextInt(NUMBER_OF_ROWS));
+                    rowItem.values[random.nextInt(rowItem.values.length)].set((100 + random.nextInt(100)) + "");
+                }
+            });
+        }
+    }
+
+    @After
+    public void afterEach() throws TimeoutException {
+        allowedToRun.set(false);
+        dataManipulators.shutdownNow();
+        FxToolkit.cleanupStages();
+    }
+
+    @Test
+    public void shouldNotFreeze_When_SelectionIsChangedUnderLoad() {
+        fillTableData();
+        produceUpdateLoad();
+
+        Duration timeTaken = measure(this::changeSelection);
+        assertThat(timeTaken, is(lessThan(MAX_DURATION_FOR_SELECTION_CHANGES)));
+    }
+
+    @Test
+    public void shouldNotFreeze_When_ScrollingVerticallyUnderLoad() {
```

Comment:
This test passes even when the fix is not applied. Not sure if its supposed to.

Reply:
Yes, it is also supposed to be that way. I first thought I could make the GUI freeze with this test, because I could see that the cells have not been reused before (8339de5). But I guess to make it freeze I need to use heavier cells, like with buttons or something like that. But as it didn't really freeze and the test was anyway written I just left it there.